riscv/sim.cc

```diff
@@ -101,7 +101,10 @@ sim_t::sim_t(const cfg_t *cfg, bool halted,
                                log_file.get(), sout_);
   }
 
-  make_dtb(dtb_file);
+  // create dts, dtb, and bootrom device
+  if (dtb_enabled) {
+    make_dtb(dtb_file);
+  }
 
   void *fdt = (void *)dtb.c_str();
```

Comment:
I'm confused how this and the following code still works if `!dtb_enabled`. It's using `fdt` which I guess points to a `'\0'` now, so `fdt_parse_clint()` et al will find nothing, so devices won't be created. In fact the entire rest of this routine will do nothing.

If I'm correct about that, then let's do something like:
```
if (!dtb_enabled) return;
make_dtb(dtb_file);
...
```

Also the commit message for f1a43675be25166ddabf7299b94cfd89bd49d45c should say something about removing devices in the case of `!dtb_enabled`.

Reply:
> I'm confused how this and the following code still works if !dtb_enabled. It's using fdt which I guess points to a '\0' now, so fdt_parse_clint() et al will find nothing, so devices won't be created. In fact the entire rest of this routine will do nothing.

Yeah that's the intent, it's more intuitive that dtb-parsed devices should not be added if `!dtb_enabled`.

I'll restructure this as you suggest.

Reply:
I also had to change some of the processor_t constructor such that cfg-passed settings would be respected even if `!dtb_enabled`. Specifically for pmpregions, and matching the paging mode for both cases.